packages/utils/src/loader/deferred.ts

```diff
@@ -12,6 +18,8 @@ export class Deferred<T> {
   private _reject: PromiseField<T> = () => {
   };
 
+  private state = PromiseState.pending;
```

Comment:
Would this better ?

`private _resolve!: (value?: T | PromiseField<T>) => void;
  private _reject!: (value?: T | PromiseField<T>) => void;
  private state = PromiseState.pending;
  private _promise: Promise<T>;

  constructor () {
    this._promise = new Promise<T>((resolve, reject) => {
      this.resolve = resolve as (value?: T | PromiseField<T>) => void;
      this.reject = reject;
    });
  }`

Reply:
Constructor looks too verbose for me. However, non-null assertion is quite helpful. Please check the updated version.